src/audio/component.c

```diff
@@ -214,3 +214,62 @@ int comp_get_copy_limits(struct comp_dev *dev, struct comp_copy_limits *cl)
 
 	return 0;
 }
+
+static void comp_update_params(uint32_t flag, struct sof_ipc_stream_params *params,
+			       struct comp_buffer *buffer)
+{
+	if (flag & BUFF_PARAMS_FRAME_FMT)
+		params->frame_fmt = buffer->frame_fmt;
+
+	if (flag & BUFF_PARAMS_BUFFER_FMT)
+		params->buffer_fmt = buffer->buffer_fmt;
+
+	if (flag & BUFF_PARAMS_CHANNELS)
+		params->channels = buffer->channels;
+
+	if (flag & BUFF_PARAMS_RATE)
+		params->rate = buffer->rate;
+}
+
+int comp_verify_params(struct comp_dev *dev, uint32_t flag,
+		       struct sof_ipc_stream_params *params,
+		       struct comp_buffer *buffer)
+{
+	struct list_item *buffer_list;
+	struct list_item *clist;
+	struct comp_buffer *sinkb;
+	struct comp_buffer *buf;
+	int dir = dev->direction;
+
+	if (!params) {
+		trace_comp_error("comp_verify_params() error: !params");
+		return -EINVAL;
+	}
+
+	if (!buffer) {
+		buffer_list = comp_buffer_list(dev, dir);
+
+		list_for_item(clist, buffer_list) {
+			buf = buffer_from_list(clist, struct comp_buffer,
+					       dir);
+
+			comp_update_params(flag, params, buf);
+
+			buffer_set_params(buf, params);
+		}
+
+		/* fetch sink buffer in order to set period frames */
+		sinkb = list_first_item(&dev->bsink_list, struct comp_buffer,
+					 source_list);
+
+		component_set_period_frames(dev, sinkb->rate);
+	} else {
+		comp_update_params(flag, params, buffer);
+
+		buffer_set_params(buffer, params);
+
+		component_set_period_frames(dev, buffer->rate);
+	}
+	
+	return 0;
```

Comment:
can you put some inline comments in here to explaining the flow and conditional logic.

Reply:
@lgirdwood 
I don't make it yet, but I will add comment and update comments